kotlin/workflow-core/src/main/java/com/squareup/workflow/StatefulWorkflow.kt

```diff
@@ -71,10 +73,15 @@ abstract class StatefulWorkflow<
    * If the workflow is being restored from a [Snapshot], [snapshot] will be the last value
    * returned from [snapshotState], and implementations that return something other than
    * [Snapshot.EMPTY] should create their initial state by parsing their snapshot.
+   * @param scope
+   * The [CoroutineScope] in which this workflow lives. The scope will be cancelled when the
+   * workflow is being torn down, so this scope can be used to start coroutines to track the
+   * lifetime of the workflow "session".
    */
   abstract fun initialState(
     input: InputT,
-    snapshot: Snapshot?
+    snapshot: Snapshot?,
+    scope: CoroutineScope
```

Comment:
@davidapgar @bencochran  @timdonnelly Seems like we should be able to come up with a similar object to pass into the analogous method in Swift — basically something that can accept tearDown duties instead of the method on WorkflowContext.

Reply:
I'm not suggesting this is a good direction, but if Swift wanted to match this as accurately as possible, it would be a teardown hook plus a scheduler.